lib/elasticsearch-filters.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+
+const RANGE_OPERATORS = ['gt', 'gte', 'lt', 'lte'];
+const TERM_MUST_OPERATORS = ['eq', 'in'];
+const TERM_MUST_NOT_OPERATORS = ['ne', 'nin'];
+
+/**
+ * @class ElasticSearchFilters
+ * @classdesc An elasticsearch filters builder
+ */
+class ElasticSearchFilters {
+
+	/**
+	 * Get the elasticsearch filters query from the recieved filters
+	 * @param {Object} filters filters
+	 * @returns {Object} elasticsearch filters
+	 */
+	static getFilters(filters) {
+
+		if(!filters || typeof filters !== 'object' || Array.isArray(filters))
+			return {};
+
+		const queries = {
+			range: this._parseRangeQueries(filters),
+			term: this._parseTermQueries(filters)
+		};
+
+		if(Object.keys(queries.range).length && Object.keys(queries.term).length) {
+			queries.term.bool.must.push(queries.range);
+			delete queries.range;
+		}
+
+		return {
+			query: { ...queries.term, ...queries.range }
+		};
+	}
+
+	/**
+	 * Prepare the filters by comparing it with the recieved operators
+	 * @param {Object} filters filters
+	 * @param {Array} operators the operators array for preparing the filters contexts
+	 * @returns {Object} prepared filters
+	 */
+	static _prepareFilters(filters, operators) {
+
+		const preparedFilters = {};
+
+		for(let [key, value] of Object.entries(filters)) {
+
+			if(!key.includes('$')) {
+				value = { [key]: value };
+				key = '$eq';
+			}
+
+			const operator = key.replace('$', '');
+
+			if(operators && operators.includes(operator))
+				preparedFilters[operator] = value;
```

Comment:
El problema con los filtros multiples de `eq` que no funcionan se podria solucionar asi:

```suggestion
				preparedFilters[operator] = { ...preparedFilters[operator], ...value };
```